ar_module2_8.js

```diff
@@ -0,0 +1,122 @@
+'use strict';
+var myDate = (function iffe() {
+  /* eslint-disable no-console */
+  /* eslint-disable no-unused-vars */
+  var privateDate = new Date(); // make this available to both get and set
+  var dd = privateDate.getDate();
+  var mm = privateDate.getMonth() + 1; //January is 0!
+  var yyyy = privateDate.getFullYear();
+
+  if (dd < 10) {
+    dd = '0' + dd;
+  }
+  if (mm < 10) {
+    mm = '0' + mm;
+  }
+
+  var checkDate = mm + dd + yyyy;
+  var formatDate = mm + '/' + dd + '/' + yyyy;
+
+  function setDate(newDate) {  // make a function for milli
+
+    if (newDate == undefined) {
+      var defaultDate = privateDate;
+      console.log(defaultDate);
+      // Done - myDate.setDate() if no argument given
+    }
+
+    else {  // only using newMilli in use
+      var milliDate = new Date();
+      milliDate.setDate(newDate);
+      console.log(milliDate);
+    }
+
+  }
+
+  function getDate(dateAsk) {
+    if (dateAsk == undefined) {
+      console.log('No Args');
+      var n = privateDate.getTime();
+      console.log(n);
+    }
+    else if (dateAsk == 'milliseconds') {
+      console.log('milliseconds');
+      var m = privateDate.getTime();
+      console.log(m);
+    }
+    else if (dateAsk == 'formatted') {
+      console.log('formatted');
+      console.log(formatDate);
+    }
+
+  }
+
+  function getDayName(dateAsk) {
+    var formatDay = privateDate.getDay();
+    var weekday = new Array(7);
```

Comment:
Whenever you need an array, it's preferable to declare array literals rather than instantiate array objects in this fashion:

`weekday = ['Sunday', 'Monday', 'Tuesday', ...];`